rustainers/src/runner/inner.rs

```diff
@@ -395,10 +513,27 @@ pub(crate) trait InnerRunner: Display + Debug + Send + Sync {
                     .await?
             }
             // Need to create and start the container
-            Some((ContainerStatus::Unknown | ContainerStatus::Removing, _)) | None => {
+            Some((ContainerStatus::Unknown | ContainerStatus::Removing, _)) => {
                 self.create_and_start(CreateAndStartOption::new(image, &options))
                     .await?
             }
+            None => {
+                // If the user has specified a network, we'll assume the user knows best
+                if options.network.is_none() & self.get_docker_host().is_none() {
+                    // Otherwise we'll try to find the docker host for dind usage.
+                    let host_network = self.find_host_network().await?;
+
+                    //Find a better alternative
+                    let mut options = options.clone();
+                    options.network = host_network;
+                    self.create_and_start(CreateAndStartOption::new(image, &options))
+                        .await?
+                } else {
+                    // If the user has specified a network, we'll assume the user knows best
+                    self.create_and_start(CreateAndStartOption::new(image, &options))
+                        .await?
```

Comment:
this duplicate line could be move out ot the if. Maybe your if expression could return the `option` to use if the original options could not be mutate inside.
Especially because `options` is used also after the  `self.create_and_start` and maybe a confusion could be introduce between steps using different `options`.